toolkit/tools/pkg/imagecustomizerlib/customizeutils.go

```diff
@@ -348,33 +350,289 @@ func enableOrDisableServices(services imagecustomizerapi.Services, imageChroot *
 		if err != nil {
-			return fmt.Errorf("failed to write module load configuration: %w", err)
+			return fmt.Errorf("failed to update disable configuration: %w", err)
 		}
 	}
 
-	for _, module := range modules.Disable {
-		logger.Log.Infof("Disabling kernel module (%s)", module.Name)
-		moduleFileName := module.Name + ".conf"
-		moduleFilePath := filepath.Join(imageChroot.RootDir(), "/etc/modprobe.d/", moduleFileName)
-		data := fmt.Sprintf("blacklist %s\n", module.Name)
-		err = file.Write(data, moduleFilePath)
+	return nil
+}
+
+func isModuleDisabled(moduleName, moduleDisableFilePath string) (bool, error) {
+	_, err := os.Stat(moduleDisableFilePath)
+	if os.IsNotExist(err) {
+		// File doesn't exist, treat as not disabled.
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+
+	file, err := os.Open(moduleDisableFilePath)
+	if err != nil {
+		return false, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		if strings.TrimSpace(scanner.Text()) == moduleName {
+			return true, nil
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		logger.Log.Errorf("Failed to scan file %s with error %s", moduleDisableFilePath, err)
+		return false, err
+	}
+
+	return false, nil
+}
+
+func removeModuleFromDisableList(moduleName, moduleDisableFilePath string) error {
+	disabled, err := isModuleDisabled(moduleName, moduleDisableFilePath)
+	if err != nil {
+		logger.Log.Infof("Error checking %s", moduleDisableFilePath)
+		return err
+	}
+
+	if disabled {
+		logger.Log.Infof("Module %s found in the disabled list. Removing...", moduleName)
+		lines, err := os.ReadFile(moduleDisableFilePath)
 		if err != nil {
-			return fmt.Errorf("failed to write module disable configuration: %w", err)
+			return err
+		}
+
+		// Filter out the line that contains the module name.
+		var updatedLines []string
+		for _, line := range strings.Split(string(lines), "\n") {
+			if !strings.Contains(line, moduleName) {
+				updatedLines = append(updatedLines, line)
+			}
 		}
+
+		return os.WriteFile(moduleDisableFilePath, []byte(strings.Join(updatedLines, "\n")), 0644)
 	}
 
 	return nil
 }
 
+func aggregateModuleOptions(aggregatedOptions []string, moduleOptionsFilePath string, moduleName string, optionKey string, optionValue string) ([]string, error) {
+	found := false
+
+	if _, err := os.Stat(moduleOptionsFilePath); err != nil {
+		if os.IsNotExist(err) {
+			// If the file does not exist, append the new option and return the options
+			return append(aggregatedOptions, fmt.Sprintf("options %s %s=%s", moduleName, optionKey, optionValue)), nil
+		}
+		return nil, err
+	}
+
+	// File exists, if the option exists in the file, directly update the file, otherwise, append it and return the options
+	file, err := os.Open(moduleOptionsFilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
```

Comment:
The files these functions are dealing with will be pretty small. So, I don't think it is necessary to use a scanner pattern here. I would just read in the entire file as a string, make the changes you want, and then write out the whole file.

Similar comment for `isModuleDisabled()`.